CMakeLists.txt

```diff
@@ -1390,6 +1390,16 @@ if(HPX_WITH_COMPILER_WARNINGS)
     hpx_add_compile_flag_if_available(-Werror=format)
     hpx_add_compile_flag_if_available(-Werror=missing-braces)
     hpx_add_compile_flag_if_available(-Werror=sign-compare)
+
+    # try to enable simd parallelization for par_unseq
+    find_package(OpenMP QUIET)
+    if(OPENMP_FOUND)
+      if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
+        hpx_add_compile_flag_if_available(-openmp-simd)
```

Comment:
`-openmp-simd` is deprecated (https://software.intel.com/en-us/node/693432).  It was replaced by `-qopenmp-simd` in version 16 or 17.

I recommend that CMake test for `-fopenmp-simd`, then `-qopenmp-simd` then `-openmp-simd`.  ICC supports many of the GCC flags for compatibility...

Reply:
Ok, thanks.